### PR TITLE
[add] Implement supported languages function

### PR DESCRIPTION
**Pull request by ZhymabekRoman**

### CODE REVIEW

**Comment by Animenosekai** [comment]
I don't know if the supported languages changes between the different endpoints (different languages for the `text to speech` and the `translate` endpoint for example)
In these cases we should have all the languages (`text to speech` and `translate` one) in the same `_supported_languages` set and verify on the `_text_to_speech` and `_translate` methods directly

File: translatepy/translators/base.py [hunk 1]
```diff
@@ -548,7 +551,13 @@ def _detect_and_validate_lang(self, language: str) -> str:
         else:
             result = Language(language)
 
-        return self._language_normalize(result)
+        normalized_result = self._language_normalize(result)
+
+        if self._supported_languages:  # Check if the attribute is not empty
+            if normalized_result not in self._supported_languages:
+                raise UnsupportedLanguage("Language {language_code} doesn't supported by {service}".format(language_code=language, service=str(self)))
```
Comment (Animenosekai) [inline translatepy/translators/base.py:558]: Typo: It should be `The language {language_code} is not supported by {service}`

**Comment by Animenosekai** [comment]
Also, as I saw it in the tests, `translatepy.exceptions.UnsupportedLanguage: Language auto doesn't supported by DeepL` was the exception catched when it failed. So I guess the `auto` language is missing from `_supported_languages` in `DeeplTranslate`

File: translatepy/translators/base.py (hunk 1 above)
Comment (ZhymabekRoman) [reply translatepy/translators/base.py:558]: Ok

**Comment by ZhymabekRoman** [comment]
> `translatepy.exceptions.UnsupportedLanguage: Language auto doesn't supported by DeepL` was the exception catched when it failed. So I guess the `auto` language is missing from `_supported_languages` in `DeeplTranslate`

auto is in the list of supported languages. It's just that I forgot that language codes must be in upper case in DeepL

**Comment by Animenosekai** [comment]
👌

**Pull request merged**